### ISSUE TITLE
v2 hangs on file with hundreds of YT links while v1 does not

### ISSUE THREAD

**Issue by PortStone** [org-roam/org-roam#1995]
### Description
I have a file called music with hundreds of links from YouTube. v1 doesn't have any problems processing it but v2 just hangs at `Processing modified files...48%`. Even after an hour of waiting it still was not finished. If I remove the file it works.
#### Steps to Reproduce
Have a file in org roam that has hundreds of YouTube links in it and try to build the database with v2 of org-roam.

The file should have entries like this (hundreds of them):
```
* Music from movies
** Heat Soundtrack - Moby - New Dawn Fades (https://youtu.be/fN5hSvu1dzw)
```
#### Expected Results

It builds just like v1 the database and I can use my notes.

#### Actual Results

It freezes at `Processing modified files...48%`.

### Environment
- Emacs: GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.27, cairo version 1.17.4)
 of 2021-03-26
- Framework: Doom
- Org: Org mode version 9.6 (9.6-??-2e99997 @ /home/portstone/.emacs.d/.local/straight/build-27.2/org/)
- Org-roam: 1.2.3

**Comment by italoaa** [comment]
having the same problem

**Comment by jethrokuan** [comment]
I've just tested with a file with a thousand links, and it took less than a second. Can you confirm if the changes in https://github.com/org-roam/org-roam/pull/1946 have been pulled in?

See also https://github.com/org-roam/org-roam/issues/1752#issuecomment-962689026

**Comment by italoaa** [comment]
I solved it by adding file by file to the folder and doing sync to the database. There was one link to a heading in the V1 way I think but everything solved thank you very much. :)

**Comment by pinacle2000** [comment]
> I've just tested with a file with a thousand links, and it took less than a second. Can you confirm if the changes in #1946 have been pulled in?
> 
> See also [#1752 (comment)](https://github.com/org-roam/org-roam/issues/1752#issuecomment-962689026)

I use doom emacs and did an update today (including the latest org-roam update), which took forever to load. I checked my config and fount it is the org roam part. When I commentted the roam part, my emacs loads in a few seconds (acceptable in Windows). I deleted the roam db and started emacs. Now it has been loading (I guess since I can only see the changing memory/cpu usage of emacs process) for half an hour, with the memory usage from dozens of MB to 322 MB now, and the number is still growing.

I have 5 hundrends of roam org files with a few levels of directory, e.g. org-roam-root/level1/level2/level3/level4/abc.org. I had 3 hundrends when I use v1 and had no such problem. Therefore I believe it is the same issue here.
